CONTRIBUTING.md

```diff
@@ -66,7 +70,11 @@ effort shall be taken to keep the length of each function limited to a maximum o
 #### Short variable names (7.1e)
 
 We place no restrictions on the length of variable names; however, recommendations against cryptic
-abbreviations and abbreviations not in a version-controlled table nor in Barr Appendix A still apply.
+abbreviations and abbreviations not in a version-controlled table still apply.
+
+#### Hungarian notation (7.1 j, k, l, m, n, o)
+
+Both Apps and Systems Hungarian notation are forbidden.
```

Comment:
Forbidden or discouraged? I'm fine either way, but I'm interested in the reasoning behind the choice of words.

Reply:
I wanted to avoid the use of the term "discouraged" except when something is usually bad, but sometimes strictly necessary. In other cases (e.g., if it is always bad, or if it is sometimes bad but never necessary), it is easier to make it strictly forbidden than to leave it open for the programmer to decide. The fewer decisions the programmer must make, the less time s/he spends thinking.

Honestly, I kind of like the idea of using Hungarian notation for pointers, but if it is strictly forbidden, then we won't have to worry about inconsistent usage of the convention.

Reply:
To me, not requiring it is enough. I agree that it is sometimes useful to call some thing a `something_ptr`, so I'm hesitant to all out forbid this. I think it we all agree Hungarian notation isn't evil per se, it's just not always necessary or useful.

Reply:
So perhaps "not required and generally discouraged" is a sensible way to put it?

Reply:
Hm, okay. It sounds like I should revert my last commit (that deletes those abbreviations) and instead just say they are never required.

Reply:
Works for me!